main/src/ca/uwaterloo/flix/language/phase/PatMatch.scala

```diff
@@ -106,64 +108,62 @@ object PatMatch {
     * @param tast The expression to check
     * @param root The AST root
     */
-  private def visitExp(tast: TypedAst.Expr)(implicit root: TypedAst.Root, flix: Flix): List[NonExhaustiveMatchError] = {
+  private def visitExp(tast: TypedAst.Expr)(implicit root: TypedAst.Root, sctx: SharedContext, flix: Flix): Unit = {
     tast match {
-      case Expr.Var(_, _, _) => Nil
-      case Expr.Hole(_, _, _, _) => Nil
+      case Expr.Var(_, _, _) => ()
+      case Expr.Hole(_, _, _, _) => ()
       case Expr.HoleWithExp(exp, _, _, _) => visitExp(exp)
       case Expr.OpenAs(_, exp, _, _) => visitExp(exp)
       case Expr.Use(_, _, exp, _) => visitExp(exp)
-      case Expr.Cst(_, _, _) => Nil
+      case Expr.Cst(_, _, _) => ()
       case Expr.Lambda(_, body, _, _) => visitExp(body)
-      case Expr.ApplyClo(exp1, exp2, _, _, _) => List(exp1, exp2).flatMap(visitExp)
-      case Expr.ApplyDef(_, exps, _, _, _, _) => exps.flatMap(visitExp)
-      case Expr.ApplyLocalDef(_, exps, _, _, _, _) => exps.flatMap(visitExp)
-      case Expr.ApplySig(_, exps, _, _, _, _) => exps.flatMap(visitExp)
+      case Expr.ApplyClo(exp1, exp2, _, _, _) => List(exp1, exp2).foreach(visitExp)
```

Comment:
I know the list was already there before, but what a strange pattern. Let us just call visitExp twice.

Reply:
The same comment about all the other List(...).foreach applies everywhere else.

Reply:
And while we are add it, we might as well format everything better with line breaks, e.g.:

```
case Expr.Blah(exp1, exp2, ..) => 
    visitExp(exp1)
    visitExp(exp2)

case Expr.Blah...
```

with some new lines thrown in